client/payments/payment-recommendations.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+	PLUGINS_STORE_NAME,
+	SETTINGS_STORE_NAME,
+	WCDataSelector,
+	Plugin,
+	WPDataSelectors,
+	OPTIONS_STORE_NAME,
+} from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import './payment-recommendations.scss';
+import { getCountryCode } from '../dashboard/utils';
+import { isWCPaySupported } from '../task-list/tasks/payments/wcpay';
+import { getAdminLink } from '../wc-admin-settings';
+import { createNoticesFromResponse } from '../lib/notices';
+
+const SEE_MORE_LINK =
+	'https://woocommerce.com/product-category/woocommerce-extensions/payment-gateways/?utm_source=payments_recommendations';
+const DISMISS_OPTION = 'woocommerce_setting_payments_recommendations_hidden';
+const SHOW_MARKETPLACE_SUGGESTION_OPTION =
+	'woocommerce_show_marketplace_suggestions';
+type SettingsSelector = WPDataSelectors & {
+	getSettings: (
+		type: string
+	) => { general: { woocommerce_default_country?: string } };
+};
+
+type OptionsSelector = WPDataSelectors & {
+	getOption: ( option: string ) => boolean | string;
+};
+
+export function getPaymentRecommendationData( select: WCDataSelector ) {
+	const { getOption, isResolving: isResolvingOption } = select(
+		OPTIONS_STORE_NAME
+	) as OptionsSelector;
+	const { getSettings } = select( SETTINGS_STORE_NAME ) as SettingsSelector;
+	const { getRecommendedPlugins } = select( PLUGINS_STORE_NAME );
+	const { general: settings = {} } = getSettings( 'general' );
+	const marketplaceSuggestions = getOption(
+		SHOW_MARKETPLACE_SUGGESTION_OPTION
+	);
+
+	const hidden = getOption( DISMISS_OPTION );
+	const countryCode = settings.woocommerce_default_country
+		? getCountryCode( settings.woocommerce_default_country )
+		: null;
+	const countrySupported = countryCode
+		? isWCPaySupported( countryCode )
+		: false;
+	const isRequestingOptions =
+		isResolvingOption( 'getOption', [ DISMISS_OPTION ] ) ||
+		isResolvingOption( 'getOption', [
+			SHOW_MARKETPLACE_SUGGESTION_OPTION,
+		] );
+
+	const canNotDisplay =
+		isRequestingOptions ||
+		hidden ||
+		marketplaceSuggestions !== 'yes' ||
+		! countrySupported;
+	let plugins;
+	if ( ! canNotDisplay ) {
+		plugins = getRecommendedPlugins( 'payments' );
+	}
+
+	// don't get recommended plugins until it is displayable.
+	return {
+		displayable: ! canNotDisplay,
+		recommendedPlugins: plugins,
+	};
+}
+
+const PaymentRecommendations = () => {
+	const [ installingPlugin, setInstallingPlugin ] = useState< string | null >(
+		null
+	);
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { installAndActivatePlugins } = useDispatch( PLUGINS_STORE_NAME );
+	const { displayable, recommendedPlugins } = useSelect(
+		getPaymentRecommendationData
+	);
+	const triggeredPageViewRef = useRef( false );
+	const shouldShowRecommendations =
+		displayable && recommendedPlugins && recommendedPlugins.length > 0;
+
+	useEffect( () => {
+		if ( shouldShowRecommendations && ! triggeredPageViewRef.current ) {
```

Comment:
This looks good. Not a required change, but seems like it'd be a great hook to introduce, because we do a lot of this kind of analytics.